[CS] fix some doc comment types

photodude opened the pull request:
Pull Request for Issue code style corrections for doc comment types.

### Summary of Changes
PHPCS2 manual fixes
- Variables passed by reference should not have the `&` prefixed in the doc comment
  - Joomla.Commenting.FunctionComment.MissingParamTag
  - Joomla.Commenting.FunctionComment.ParamNameNoMatch
- Correct return statement

[Found using the Joomla code standards 2.0.0 PHPCS2-RC](https://github.com/joomla/coding-standards/releases/tag/2.0.0-rc)

### Testing Instructions
Merge by code review

### Expected result
code style has been applied as listed above, old code style testing on drone does not error.

### Actual result
code style had not been applied. [Found using the Joomla code standards 2.0.0 PHPCS2-RC](https://github.com/joomla/coding-standards/releases/tag/2.0.0-rc)


### Documentation Changes Required
none

photodude committed:
PHPCS2 manual fixes

- Variables passed by reference should not have the `&` prefixed in the doc comment
  - Joomla.Commenting.FunctionComment.MissingParamTag
  - Joomla.Commenting.FunctionComment.ParamNameNoMatch
- correct return statement

zero-24 commented:
Can be merged when 3.8.8 is released. Thanks @photodude

photodude commented:
@zero-24 Looks like the PHPCS 1.5.x code standards on Drone does not like these changes...

I'm not sure which way is the correct way. I'm guessing we need to fix something in PHPCS 2.x sniffs

mbabker commented:
You might need a custom exemption for the CMS for now to ignore that case.  Because the PHPCS 1.5 sniffs enforce the reference operator being in the doc block and 2.0+ enforce it not being there.

photodude commented:
@mbabker Is it correct in our code style to enforce the reference operator being in the doc block or to enforce it not being there? Reading over the docs I can't find any reference indicating which way is correct.

mbabker commented:
I don't think we ever decided one way or the other, I've just went with what the tooling told me to do.  Arguably it's more correct to not have it, it's not necessary to describe in the doc block that something is passed by reference (that should be used in a code parser).

photodude commented:
Doing some reading in other places I came across this comment in a phpDocumentor issue: "Having a "&" in the docblock (whether it's in front of the type or name) is not valid. References are to be expressed solely by the function/method declaration." 

So it seems our PHPCS 1.5 sniffs are wrong and our PHPCS 2.x+ sniffs are correct.

I'll look at adding the exceptions here

photodude committed:
exclude administrator/components/ folder

"Having a "&" in the docblock (whether it's in front of the type or name) is not valid. References are to be expressed solely by the function/method declaration."

so we need to exclude this sniff while we trasition to the PHPCS 2.x version

carlitorweb commented:
This warning is only for me, or the phpcs have a problem too in this. Almost all files I open, have it
![screenshot_20180522082902](https://user-images.githubusercontent.com/3956472/40362304-2321c9ce-5d9a-11e8-8b39-01ea6646f4fc.png)

photodude commented:
@carlitorweb please file that with [PHPCS](https://github.com/squizlabs/PHP_CodeSniffer) that error comes from the core PHPCS generic sniffs. I have seen it on some files when running PHPCS on win10 but I have not looking into it nor tried fixing files automatically that report it.

carlitorweb commented:
Ok sorry then, and thank you @photodude

photodude commented:
No problem @carlitorweb, If you find any issues with the Joomla custom sniffs for PHPCS2.x+ we also have a repo specific to those sniffs where they can be reported. https://github.com/joomla/coding-standards
